waspc/packages/deploy/src/providers/railway/railwayService/url.ts

```diff
@@ -0,0 +1,53 @@
+import { WaspProjectDir } from "../../../common/brandedTypes.js";
+import { createCommandWithCwd } from "../../../common/zx.js";
+import {
+  ClientServiceName,
+  Port,
+  RailwayCliExe,
+  ServerServiceName,
+} from "../brandedTypes.js";
+import { RailwayCliDomainSchema } from "../jsonOutputSchemas.js";
+
+export enum ServiceUrlStatus {
+  URL_CREATED = "URL_CREATED",
+  URL_ALREADY_EXISTS = "URL_ALREADY_EXISTS",
+}
+
+export async function generateServiceUrl(
+  serviceName: ClientServiceName | ServerServiceName,
+  port: Port,
+  options: {
+    railwayExe: RailwayCliExe;
+    waspProjectDir: WaspProjectDir;
+  },
+): Promise<string> {
+  const railwayCli = createCommandWithCwd(
+    options.railwayExe,
+    options.waspProjectDir,
+  );
+  const result = await railwayCli(
+    ["domain", "--service", serviceName, "--port", port.toString(), "--json"],
+    {
+      nothrow: true,
+    },
+  );
+
+  if (result.exitCode !== 0) {
+    throw new Error(`There was a problem getting a domain for ${serviceName}.`);
+  }
+
+  if (result.stdout.includes("Domains already exists on the service:")) {
+    return matchServiceUrl(result.stdout);
+  } else {
+    const { domain } = RailwayCliDomainSchema.parse(result.json());
+    return domain;
+  }
+}
+
+function matchServiceUrl(text: string): string {
```

Comment:
I know this is helper functoin, but it is overly mysterious, with the "match" wihch sound slike implementation detail, and not clear what the input is, what is output, ... . I would say either inline it, or make it more standalone (name, comments, something).